contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,426 @@
+            revert Errors.LicenseRegistry__ZeroDisputeModule();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.disputeModule = IDisputeModule(newDisputeModule);
+    }
+
+    /// @dev Sets the LicensingModule address.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param newLicensingModule The address of the LicensingModule
+    function setLicensingModule(address newLicensingModule) external onlyProtocolAdmin {
+        if (newLicensingModule == address(0)) {
+            revert Errors.LicenseRegistry__ZeroLicensingModule();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.licensingModule = ILicensingModule(newLicensingModule);
+    }
+
+    /// @notice Sets the default license terms that are attached to all IPs by default.
+    /// @param newLicenseTemplate The address of the new default license template.
+    /// @param newLicenseTermsId The ID of the new default license terms.
+    function setDefaultLicenseTerms(address newLicenseTemplate, uint256 newLicenseTermsId) external onlyProtocolAdmin {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.defaultLicenseTemplate = newLicenseTemplate;
+        $.defaultLicenseTermsId = newLicenseTermsId;
+    }
+
+    /// @notice Registers a new license template in the Story Protocol.
+    /// @param licenseTemplate The address of the license template to register.
+    function registerLicenseTemplate(address licenseTemplate) external onlyProtocolAdmin {
+        if (licenseTemplate.supportsInterface(type(ILicenseTemplate).interfaceId)) {
+            revert Errors.LicenseRegistry__NotLicenseTemplate(licenseTemplate);
+        }
+        _getLicenseRegistryStorage().registeredLicenseTemplates[licenseTemplate] = true;
+        emit LicenseTemplateRegistered(licenseTemplate);
+    }
+
+    /// @notice Registers a new royalty policy in the Story Protocol.
+    /// @param royaltyPolicy The address of the royalty policy to register.
+    function registerRoyaltyPolicy(address royaltyPolicy) external onlyProtocolAdmin {
+        _getLicenseRegistryStorage().registeredRoyaltyPolicies[royaltyPolicy] = true;
+        emit RoyaltyPolicyRegistered(royaltyPolicy);
+    }
+
+    /// @notice Registers a new currency token used for paying license token minting fees and royalties.
+    /// @param token The address of the currency token to register.
+    function registerCurrencyToken(address token) external onlyProtocolAdmin {
+        _getLicenseRegistryStorage().registeredCurrencyTokens[token] = true;
+        emit CurrencyTokenRegistered(token);
+    }
+
+    /// @notice Sets the expiration time for an IP.
+    /// @param ipId The address of the IP.
+    /// @param expireTime The new expiration time, 0 means never expired.
+    function setExpireTime(address ipId, uint256 expireTime) external onlyLicensingModule {
+        _setExpireTime(ipId, expireTime);
```

Comment:
do you need to check expireTime is greater than current?

Reply:
As the function can only called by LicensingModule, we have to make sure this condition is in the module.